WIP: Convertaxis vector rebased

scottwittenburg opened the pull request.

scottwittenburg commented:
We are down to a single test failing, `tests/test_vcs_axisconvert.py`, and it only seems to fail on Python 3.  Which means that so far I can't reproduce it.  I'll need to set up a cdat conda environment for Python 3 to see what's going on.

scottwittenburg commented:
@danlipsa @charlesdoutriaux I think this is ready for you guys to start taking a look.

danlipsa commented:
Baselines in
https://github.com/CDAT/uvcdat-testdata/pull/194

doutriaux1 commented:
@scottwittenburg thanks for this. I will review today as I'm back from vacation.

danlipsa commented on vcs/VTKPlots.py:
```diff
@@ -1428,16 +1442,77 @@ def scaleLogo(self):
-    def fitToViewport(self, Actor, vp, wc=None, geoBounds=None, geo=None, priority=None,
-                      create_renderer=False, add_actor=True):
+    def _applyTransformationToDataset(self, T, data):
+        global index
```
We need to remove the index and debugWriteGrid - this was only for debugging purposes.

danlipsa commented on vcs/vcs2vtk.py:
```diff
@@ -15,6 +15,27 @@
+        writer.SetInputData(grid)
+        writer.Write()
+
+
```
Lets remove this.

danlipsa commented on vcs/vcs2vtk.py:
```diff
@@ -579,6 +600,9 @@ def genGrid(data1, data2, gm, deep=True, grid=None, geo=None, genVectors=False,
     vg.GetCellData().SetGlobalIds(globalIds)
+
+    debugWriteGrid(vg, "vg")
+
```
Remove

danlipsa commented on vcs/vcs2vtk.py:
```diff
@@ -1418,6 +1442,9 @@ def prepFillarea(context, renWin, farea, cmap=None):
     polygonPolyData.SetPoints(pts)
+
+    debugWriteGrid(polygonPolyData, "fillarea")
+
```
Remove

danlipsa commented on vcs/vcsvtk/boxfillpipeline.py:
```diff
@@ -120,14 +119,16 @@ def _plotInternal(self):
+                    # transformFilter.SetInputData(mapper.GetInput())
+                    # transformFilter.SetTransform(transform)
+                    # transformFilter.Update()
 
```
Lets remove this as it is commented out.

danlipsa commented on vcs/vcsvtk/isofillpipeline.py:
```diff
@@ -147,14 +147,17 @@ def _plotInternal(self):
+                # transformFilter.Update()
+
+                # patact = fillareautils.make_patterned_polydata(transformFilter.GetOutput(),
+
```
Lets remove this as it is commented out.

danlipsa commented on vcs/vcsvtk/meshfillpipeline.py:
```diff
@@ -205,13 +205,15 @@ def _plotInternal(self):
+                    # transformFilter.SetInputData(mapper.GetInput())
+                    # transformFilter.SetTransform(transform)
+                    # transformFilter.Update()
+
```
Lets remove this as it is commented out.

scottwittenburg commented:
Thanks for taking a look @danlipsa.  Where does the sentence describing what marker size means belong?  In `vcs/marker.py`?  Or in `vcs/xmldocs.py`?

scottwittenburg commented:
@danlipsa Do you think this could be ready to merge now?

danlipsa commented:
+1 LGTM. @doutriaux1 Any comments?

scottwittenburg commented:
pinging @aashish24 for visibility

scottwittenburg commented:
ping @sankhesh

doutriaux1 commented:
@scottwittenburg sorry I'm still trying to get to this!

sankhesh commented:
LGTM :+1:

doutriaux1 commented on tests/test_vcs_axisconvert_2d.py:
```diff
@@ -55,9 +70,12 @@ def test_areawt(self):
-                self.axisConvertGmLog10Areawt(method)
+                self.axisConvertGmLinear(method)
+                if not method is vcs.createstreamline:
+                    self.axisConvertGmLog10(method)
```
@danlipsa @scottwittenburg not log10 for streamlines? I guess it doesn't make much sense anyway. But just wondering.

scottwittenburg replied:
In general it should be fine whenever log10 scaling the axis makes sense, it just didn't seem to make sense with the test data sets (we can't do log10 on an axis when there are negative values there, and it doesn't make sense when the two variables are the same).  Is there a better pair of variables for this particular test case?  Same situation for vectors really I guess.

scottwittenburg commented:
@charlesdoutriaux @danlipsa In our recent email discussion, one thing I said I would do is increase the default marker size for all plots in this branch.  The reason was because the default size now results in markers that are not visible.  I just want to show an example of how some plots using the current default marker size of 1 will change.

![default_marker_size_change](https://user-images.githubusercontent.com/6527504/38887674-c16d490c-4236-11e8-9036-09d862b2a317.png)

In the image above, I have changed the default marker size from 1 to 10.  As you can see, the markers with size 1 are not completely invisible, just small as they're only 1px big.  Is 10 the new default value we want?

scottwittenburg commented:
Or maybe we would prefer a default size of around `5px` instead?

scottwittenburg commented:
@charlesdoutriaux @danlipsa  Just let me know what default marker size we want, so I can make the change and push a bunch more baselines.

doutriaux1 commented:
@scottwittenburg the right size would be the new default? Is this on the branch where we change the default to be 10 (from 1) or is this the new default for size=1 ?

danlipsa commented:
Probably 5 pixels would look fine as well. I don't really have a preference between 10 and 5.

doutriaux1 commented:
i like 10. I think it makes for easy maths for end user as well

scottwittenburg commented:
So this is the original topic, where I have now changed the default marker size to be 10, and updated the baselines to match (fixing a couple of variable number changes in the process).  The tests are passing, so I think it's all set for comparison against the ["10x scaling" branch](https://github.com/CDAT/vcs/pull/334).

@danlipsa If you want to take another look, and then if you're ok with everything, maybe mark your current review as resolved, that will remove the final red mark from this topic :smile: 

ping @charlesdoutriaux @aashish24

aashish24 commented:
> Not yet, I'm still trying to get users feedbacks on the two possible ones.

If I may suggest merging these and if users prefer one or the other.. remove or disable that part of code if possible. I would think you want users to try out master and not individual branches. This is my thinking based on the my experiences.
